compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -2099,12 +2105,9 @@ void generateNewTypedescIns(BIRNonTerminator.NewTypeDesc newTypeDesc) {
     }
 
     private boolean isNonReferredRecord(BType type) {
```

Comment:
Isn't this now a check for a module-level record? Please update the name appropriately.

Reply:
Updated

Reply:
Still the same name?

Reply:
Introduce a separate function called `isUserDefinedTypeDefinition` in the `Types` class. Updated